internal/service/service.go

```diff
@@ -0,0 +1,180 @@
+	return nil
+}
+
+func (t Serv) GetPositions(limit, offset int) ([]internal.Position, error) {
+	m := t.repo.GetPositions()
+	positions := make([]internal.Position, 0)
+	for _, value := range m {
+		positions = append(positions, value)
+	}
+	offset--
+	if float64(len(positions))/float64(limit) < float64(offset) || limit < 1 || offset < 0 {
+		return nil, errors.New("incorrect data")
+	}
+	var answer []internal.Position
+	if len(positions) == 0 {
+		return answer, nil
+	}
+	for i := limit * offset; i < limit*offset+limit && i < len(positions); i++ {
+		answer = append(answer, positions[i])
+	}
+	return answer, nil
+}
+
+func (t Serv) GetEmployees(limit, offset int) ([]internal.Employee, error) {
+	m := t.repo.GetEmployees()
+	employees := make([]internal.Employee, 0)
+	for _, value := range m {
+		employees = append(employees, value)
+	}
+	offset--
+	if float64(len(employees))/float64(limit) < float64(offset) || limit < 1 || offset < 0 {
+		return nil, errors.New("incorrect data")
+	}
+	var answer []internal.Employee
+	if len(employees) == 0 {
+		return answer, nil
+	}
+	for i := limit * offset; i < limit*offset+limit && i < len(employees); i++ {
+		answer = append(answer, employees[i])
+	}
+	return answer, nil
+}
+
+func (t Serv) GetPosition(id string) (internal.Position, error) {
+	m := t.repo.GetPositions()
+	uID, err := uuid.Parse(id)
+	if err != nil {
+		return internal.Position{}, err
+	}
+	for _, value := range m {
+		if value.ID == uID {
+			return value, nil
+		}
+	}
+	return internal.Position{}, errors.New("not found")
+}
+
+func (t Serv) GetEmployee(id string) (internal.Employee, error) {
+	m := t.repo.GetEmployees()
+	uID, err := uuid.Parse(id)
+	if err != nil {
+		return internal.Employee{}, err
+	}
+	for _, value := range m {
+		if value.ID == uID {
+			return value, nil
+		}
+	}
+	return internal.Employee{}, errors.New("not found")
+}
+
+func (t Serv) DeletePosition(id string) error {
+	m := t.repo.GetPositions()
+	uID, err := uuid.Parse(id)
+	if err != nil {
+		return err
+	}
+	for key, value := range m {
+		if value.ID == uID {
+			delete(m, key)
+		}
+	}
+	return errors.New("not found")
+}
+
+func (t Serv) DeleteEmployee(id string) error {
+	m := t.repo.GetEmployees()
+	uID, err := uuid.Parse(id)
+	if err != nil {
+		return err
+	}
+	for key, value := range m {
+		if value.ID == uID {
+			delete(m, key)
+		}
+	}
+	return errors.New("not found")
+}
+
+func (t Serv) UpdatePosition(p *internal.Position) error {
+	m := t.repo.GetPositions()
+	if p.ID.String() == uuid.Nil.String() {
+		return errors.New("incorrect data")
+	}
+	if _, ok := m[p.ID.String()]; ok {
+		delete(m, p.ID.String())
+		t.repo.AddPosition(p)
+	}
+	return errors.New("not found")
+}
+
+func (t Serv) UpdateEmployee(e *internal.Employee) error {
+	m := t.repo.GetEmployees()
+	fmt.Println(e.ID.String())
+	if e.ID.String() == uuid.Nil.String() {
+		return errors.New("incorrect data")
+	}
+	if _, ok := m[e.ID.String()]; ok {
+		delete(m, e.ID.String())
```

Comment:
Please try to run service and check how the method works. 
This call is useless because here you are working with local copy of array. 
You should perform delete operation through repo call

Reply:
+